### DISCUSSION TITLE
Open ports for DMS hosted on local system? (Fritz!Box)

### DISCUSSION THREAD

**Discussion by SchBenedikt** [docker-mailserver/docker-mailserver#3831] [Q&A] [answered]
### Description

I've been able to install everything so far, the local configuration also works.

I would now like to make everything accessible from the Internet, with a domain, which is why I adapted the code so that everything works. (So domain name, etc. adjusted.)

- I have also released the ports with the Fritz!Box, but unfortunately they are not yet open.
- How can I open the ports?
- Can I open it in Crowdsec and or Cloudflare?

Thanks for the help,
schBenedikt

**Comment by georglauterbach** [comment, marked as answer]
I assume you are hosting this on your home network. There are many downsides to that; I'd highly advise against it, and here is why:

1. Probably no Static IP (or no IPv4 even, only IPv6):
   - when using normal internet connections (at least in Germany), you will not get a static IP address
   - this will cause trouble when your IP changes (with DSL maybe once a day), but DNS caches have not yet updated, which causes a A-PTR-mismatch in DNS
   - result: many mail servers will reject mail (mine would)
2. "Just opening ports" does not work
   - NAT aside, the device on which DMS is hosted will need its port open too
   - NAT not aside, you will have trouble identifying the origin IP (because in the local network, mail appears to be coming from your Fritz!Box, always
      - result: this renders anti-spam measures almost completely useless
   - you'd want to use the [PROXY protocol](https://docker-mailserver.github.io/docker-mailserver/latest/examples/tutorials/mailserver-behind-proxy/) in case the DMS host has no [routable IP address](https://www.skillset.com/questions/describe-the-difference-between-a-routable-and-non-routable-ip-address-choose-all-that-apply); I don't know whether FritzBox supports PROXY
3. Whether your ISP allows incoming traffic on relevant ports (e.g. 25) is not guaranteed (unlikely)
   - result: when your ISP blocks relevant ports, you can immediately forget about hosting at home anyway
 4. You will probably be on denylists from the beginning
    - let's assume your ISP is Vodafone; your IP address consequently belongs to Telekom; because denylist providers know that dynamic IPs (see point 1) are not nice to deal with, they will incur a penalty for using such an IP right away
    - result: when using services like SpamHaus, you may be flagged as spam right away

My advice: use a VPS with a static IPv4 address.

---

This is not suited towards an issue; I will convert it to a discussion.
